doc/serial-timeout-troubleshoot.md

```diff
@@ -0,0 +1,80 @@
+# Troubleshooting "serial timeout" errors
+
+Several nRF Connect Desktop applications use serial ports to communicate with
+development kits and dongles. nRF Connect Desktop also uses serial ports to
+perform DFU (Device Firmware Update) when a devkit/dongle is running a 
+[DFU bootloader](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/sdk_app_serial_dfu_bootloader.html?cp=4_0_0_4_3_4).
+
+Serial ports might be [UARTs](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52840.ps/uart.html?cp=2_0_0_5_32)
+on the devkits connected to a RS-232 port on a computer,
+or they might be emulated through USB CDC ACM (also known as "Virtual COM port",
+"serial over USB" or other names). The later happens when using 
+[USB ACM CDC on the IMCU](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52/dita/nrf52/development/preview_dev_kit/vir_com_port.html?cp=2_1_5_4_1),
+or when using [USB functionality from the Nordic SDK](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/lib_usbd_class_cdc.html?cp=4_0_0_3_51_8_3)
+on SoCs that implement USB (e.g. the nRF52840).
+
+While serial ports are ubiquitous, serial port transport lacks any protocol
+metadata or capabilities for request-response higher-level protocols. Combined
+with the fact that modern desktop operating systems still allow for legacy
+serial ports, this can trigger timeouts.
+
+This document collects some known scenarios and workarounds for these timeout 
+errors.
+
+# Scenarios where these errors happen
+
+
+## DevKit/dongle not running bootloader
+
+When using a nRF52840 devkit/dongle and performing a DFU operation with 
+nRF Connect Desktop, a `Timeout while reading from serial transport.` error shows up.
+
+When using a nRF52840 devkit/dongle and performing a DFU operation with
+[nrfutil](https://github.com/NordicSemiconductor/pc-nrfutil/), a 
+`TypeError: 'NoneType' object is not iterable` error shows up.
+
+When performing a DFU operation, nrfutil *assumes* that the serial port
+specified is a devkit/dongle already in bootloader mode. nRF Connect Desktop (and
+the underlying [`nrf-device-setup-js`](https://github.com/NordicSemiconductor/nrf-device-setup-js) library)
+will try to perform a [DFU trigger](http://infocenter.nordicsemi.com/topic/com.nordic.infocenter.sdk5.v15.0.0/lib_dfu_trigger_usb.html) 
+first, but the DFU trigger might fail due to other reasons.
+
+In either case, the firmware in the nRF SoC is receiving DFU data through the
+serial port, but is not understanding it, and it's not replying to it.
+
+#### Workaround
+
+Ensure that the devkit/dongle is in bootloader mode.
+
+- For nRF52840 dongles:
+    - Unplug it from its USB port
+    - Hold down the reset button.
+    - Plug the dongle into a USB port. The red LED should be pulsing, indicating the nRF device is in DFU bootloader mode.
+    - Stop holding down the reset button.
```

Comment:
Do we have to  hold down the reset button and plug in? As far as I experienced, plugging in and clicking the reset button once works for me. I feel it is easier than holding a button and plug in.

Reply:
IIRC, the "hold the reset button down while plugging the thing in" is how @SebastianBoe and @frkv designed the bootloader firmware.

I'll let the current wording stay, unless I can get confirmation from @frkv that pressing down the reset button during run-time will always switch into bootloader mode.

Reply:
The nRF52840 Dongle User Guide states that you can enter bootloader mode by pressing reset button. There is no mention of replugging the device while holding the button.

Also, it is correct that other JLink based Nordic kits behave differently, though for those devices replugging while holding the button will put the *JLink* MCU in bootloader mode.

Reply:
Initially this was the design idea was to have a purposeful reset button. It seems that the design has changed a little bit so the requirement to power cycle while holding is no longer necessary. Simple press is all that is required. I think that this change is a more convenient way of handling going into bootloader mode, but the thing we lost in the process was the purposeful reset-button on the device which have to be solved by power cycle or expressive GPIO usage

Reply:
I confirmed the behaviour with @frkv and Øyvind. Will change the wording.